docs/admin/kubelet-tls-bootstrapping.md

```diff
@@ -0,0 +1,96 @@
+The API for requesting certificates adds a certificate-issuing control loop to the KCM. This takes the form of a 
+[cfssl](https://blog.cloudflare.com/introducing-cfssl/) local signer using assets on disk. 
+Currently, all certificates issued have one year validity and a default set of key usages. 
+
+### Signing assets 
+You must provide a Certificate Authority in order to provide the cryptographic materials necessary to issue certificates. 
+This CA should be trusted by the apiserver for authentication with the `--client-ca-file=SOMEFILE` flag. The management
+of the CA is beyond the scope of this document but it is recommended that you generate a dedicated CA for Kubernetes.
+Both certificate and key are assumed to be PEM-encoded. 
+
+The new controller-manager flags are: 
+```
+--cluster-signing-cert-file="/etc/path/to/kubernetes/ca/ca.crt" --cluster-signing-key-file="/etc/path/to/kubernetes/ca/ca.key"
+``` 
+
+### Auto-approval 
+To ease deployment and testing, the alpha version of the certificate request API includes a flag to approve all certificate 
+requests made by users in a certain group. The intended use of this is to whitelist only the group corresponding to the bootstrap
+token in the token file above. Use of this flag circumvents makes the "approval" process described below and is not recommended
+for production use. 
+
+The flag is: 
+```
+--insecure-experimental-approve-all-kubelet-csrs-for-group="system:kubelet-bootstrap"
+``` 
+
+## kubelet configuration 
+To use request a client cert from the certificate request API, the kubelet needs a path to a kubeconfig file that contains the
+bootstrap auth token. If the file specified by `--kubeconfig` does not exist, the bootstrap kubeconfig is used to request a
+client certificate from the API server. On success, a kubeconfig file referencing the generated key and obtained certificate
+is written to the path specified by `--kubeconfig`. The certificate and key file will be stored in the directory pointed 
+by `--cert-dir`.  The new flag is: 
+
+```
+--experimental-bootstrap-kubeconfig="/path/to/bootstrap/kubeconfig"
+``` 
+
+## kubectl approval 
+The signing controller does not immediately sign all certificate requests. Instead, it waits until they have been flagged with an 
+"Approved" status by an appropriately-privileged user. This is intended to eventually be an automated process handled by an external
+approval controller, but for the alpha version of the API it can be done manually by a cluster administrator using kubectl. 
+An administrator can list CSRs with `kubectl get csr`, describe one in detail with `kubectl describe <name>`. There are 
+[currently no direct approve/deny commands](https://github.com/kubernetes/kubernetes/issues/30163) so an approver will need to update
+the Status field directly. A rough example of how to do this in bash which should only be used until the porcelain merges is available
+at https://github.com/gtank/csrctl.
```

Comment:
Consider link-ifying that url:

    at [https://github.com/gtank/csrctl](https://github.com/gtank/csrctl)